response.go

```diff
@@ -14,8 +14,9 @@ import (
 // A Response is Typhon's wrapper around http.Response, used by both clients and servers.
 type Response struct {
 	*http.Response
-	Error   error
-	Request *Request // The Request that we are responding to
+	Error    error
+	Request  *Request // The Request that we are responding to
```

Comment:
`Request` is also exported as part of `*http.Response`. It isn't immediately clear to me why we need this to be `*typhon.Request` here?

Reply:
The `net/http` documentation [states](https://golang.org/pkg/net/http/#Response):
>This is only populated for Client requests.

Typhon populates this field regardless of whether it is handling the request as a client or a server. Also, users of Typhon find it more convenient to use its wrapper types.